packages/sdk/src/protocols/filter.ts

```diff
@@ -245,12 +244,12 @@ class FilterSDK extends BaseProtocolSDK implements IFilterSDK {
    * @returns The subscription object.
    */
   async createSubscription(
-    pubsubTopicShardInfo: SingleShardInfo | PubsubTopic = DefaultPubsubTopic
+    pubsubTopicShardInfo: ShardingParams | PubsubTopic
   ): Promise<SubscriptionManager> {
     const pubsubTopic =
       typeof pubsubTopicShardInfo == "string"
         ? pubsubTopicShardInfo
-        : singleShardInfoToPubsubTopic(pubsubTopicShardInfo);
+        : shardInfoToPubsubTopics(pubsubTopicShardInfo)?.[0];
```

Comment:
`ShardingParams` can be mapped to many `pubsubTopics` - we provide subscription to one. 

Any idea how to handle it better?

Reply:
@weboko we could forbid passing shard info where `shards` length is greater than 1